validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -180,19 +196,25 @@
   private final ExceptionMap exceptionMap;
   private final String generation;
   private final List<DeviceCredential> deviceCredentials = new ArrayList<>();
+  private final TreeMap<String, Object> siteMetadata;
```

Comment:
Seems like it would be cleaner overall to keep the class variable as Metadata (rather than TreeMap), and then convert to TreeMap only at the point of use?

Reply:
Then we convert it once per metadata loaded?

Reply:
True -- although I've never seen performance be an actual issue with any of this stuff :-). I also think it's only likely to be called relatively infrequently per device (once, maybe twice?)  Your call, I'm good either way.